keras/src/layers/preprocessing/discretization.py

```diff
@@ -96,8 +96,7 @@ def __init__(
         name=None,
     ):
         if dtype is None:
-            dtype = "int64" if output_mode == "int" else backend.floatx()
-
+            dtype = "float32"
```

Comment:
I think you should just remove the whole `if dtype is None` block, the base layer class will handle `None` properly.

Reply:
I tried that! It results in the unit tests failing 
```
FAILED keras/src/layers/preprocessing/discretization_test.py::DiscretizationTest::test_discretization_basics - AssertionError: expected output dtype int64, got int32:
- int32
+ int64
```